RE: Delete PrimaryHeroImages files when instance is deleted

xlisachan opened the pull request:
Fixes #14729 

The following updates have been made with this pull request:
- Image files are deleted upon deleting PrimaryHeroImage instance
- `discovery/tests/test_admin.py` has been updated

I had tried to place what's in `models.py` in `admin.py` (changing `self` to `obj`, etc) and experimented with `delete_model` as well. However, I'd either (1) get an error (`'basequeryset' has not attribute custom_image`) and the files would not delete while all tests in `test_admin`would pass, or (2) the images would get deleted upon deleting the instance, but the tests would fail. 🤨 Please review when you have a moment. Thanks!

eviljeff commented:
> I had tried to place what's in models.py in admin.py (changing self to obj, etc) and experimented with delete_model as well.

I think it's better in the model's `delete()` anyway - we _always_ want the files to be tidied up, even if the delete happens through some other means (example: maybe in the future they'll be an API to delete them)

eviljeff commented on src/olympia/hero/models.py:
```diff
@@ -113,6 +113,14 @@ def preview_image(self):
 
+    def delete(self, *args, **kwargs):
+        (path, fn) = os.path.split(self.custom_image.path)
+        dest_thumb = path + b'/thumbs/' + fn
```
I think as we're doing this in two places now (both here and in `PrimaryHeroImage.save_model`) we could do with abstracting it out into an `@property` (`def thumbnail_path`?) and then using that here and in `save_model`.

xlisachan commented:
Thanks @eviljeff! Per your feedback, please see the changes made below:
- Imported `default_storage as storage` and replaced `os.remove(...)` to `storage.delete(...)`
- Added a check to see if the files existed in storage with `storage.exists(...)`
- Added `thumbnail_path` and updated `save.model` and `delete`

Please review when you have a moment. Thank you!
